src/main/java/net/mcreator/ui/component/UnsupportedComponent.java

```diff
@@ -23,20 +23,65 @@
 
 import javax.swing.*;
 import java.awt.*;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.util.Arrays;
 
 public class UnsupportedComponent extends JPanel {
 
+	/**
+	 * Helper util method that marks provided component as not supported.
+	 *
+	 * @param comp The component to be marked.
+	 */
+	public static void markUnsupported(Component comp) {
+		Container parent = comp.getParent();
+		if (parent != null) {
+			int index = Arrays.asList(parent.getComponents()).indexOf(comp);
+			parent.remove(index);
+			parent.add(new UnsupportedComponent(comp), index);
+		}
+	}
+
 	private final Image warning = UIRES.get("18px.warning").getImage();
 
+	/**
+	 * Constructs a panel that displays an overlay with "unsupported" warning over the component marked as such.
+	 *
+	 * @param origin The component to be marked.
+	 */
 	public UnsupportedComponent(Component origin) {
 		setLayout(new GridLayout());
-
-		if (!(origin instanceof JSpinner))
-			add(origin);
-
 		setOpaque(false);
 
+		// disable origin component and prevent any mouse clicks/key presses from being handled by it
 		origin.setEnabled(false);
+		origin.addMouseListener(new MouseAdapter() {

```

Comment:
This will not guarantee events are prevented, as listeners registered before this one are still processed before the event is consumed. I guess you would need to get all listeners list, remove them, add this one first and then add other listeners back.

Reply:
Maybe just remove all the listeners instead? As the component is disabled, it should not trigger any actions at all I could say.

Reply:
That is of course a valid point. You can do that indeed

Reply:
✔️ Fixed